h5py/_hl/filters.py

```diff
@@ -358,7 +358,7 @@ def guess_chunk(shape, maxshape, typesize):
 
     # Determine the optimal chunk size in bytes using a PyTables expression.
     # This is kept as a float.
-    dset_size = np.product(chunks)*typesize
+    dset_size = np.prod(chunks)*typesize
```

Comment:
I would move these over too?  I think chunks are still shape-like.

Not going to block merging over.

Reply:
Your call, I just didn't want to optimize code that I didn't immediately understand :)

Reply:
done ! FWIW, I measured that `h5py._hl.base.product` was indeed faster than `np.prod` on small arrays such as `chunks`